xbmc/guilib/GUIEditControl.cpp

```diff
@@ -553,9 +553,16 @@ bool CGUIEditControl::SetStyledText(const std::wstring &text)
   unsigned int startSelection = m_cursorPos + m_editOffset;
   unsigned int endSelection   = m_cursorPos + m_editOffset + m_editLength;
 
+  unsigned int style = 0;
+  CGUIFont* font = m_label2.GetLabelInfo().font;
+  if (font)
+  {
+    style = font->GetStyle() << 24;
```

Comment:
can you explain me what means the shift to 24?

Reply:
The font style bits are stored in the upper byte in vecText. 

```
float CGUITextLayout::GetTextWidth(const std::wstring &text) const
{
  // NOTE: Assumes a single line of text
  if (!m_font) return 0;
  vecText utf32;
  AppendToUTF32(text, (m_font->GetStyle() & FONT_STYLE_MASK) << 24, utf32);
  return m_font->GetTextWidth(utf32);
}
```

Another side effect of this change is that the text entered into the edit box should now match the font style of the theme (which I believe was the original intent).